### DISCUSSION TITLE
Is there a way to diff 2 seperate MetaData obejct?

### DISCUSSION THREAD

**Discussion by dvaerum** [sqlalchemy/sqlalchemy#6340] [Usage Questions]
I have a MetaData object for a Database, created by using `reflect`
```python
meta_old = MetaData()
meta_old.reflect(bind=engine)
```
and I would like to know how it is different from the one I build myself using `declarative_base()`
```
Base = declarative_base()
meta_new = Base.metadate
```

**Comment by zzzeek** [comment]
your best bet is to use alembic's feature , see the example at https://alembic.sqlalchemy.org/en/latest/api/autogenerate.html#alembic.autogenerate.compare_metadata .  you would skip the meta.reflect() part, it does it's own thing internally.

**Reply by dvaerum** [reply]
That was also what I have found, but for in to work in this situation I have to build a DB before I can compare it with metadata, but that is a possible workaround, I was just hoping to be able to compare 2 MetaData objects.

Anyway Thanks for the feedback 😃
